autoscale/src/main/java/com/sequenceiq/periscope/controller/AutoScaleClusterCommonService.java

```diff
@@ -259,4 +274,11 @@ protected void createAutoscalingConfigChangedHistoryAndNotify(Cluster cluster) {
         notificationSender.sendHistoryUpdateNotification(
                 historyService.createEntry(scalingStatus, statusMessage, cluster), cluster);
     }
+
+    private boolean newOrPreExistingTimeAlerts(Cluster cluster, DistroXAutoscaleClusterRequest distroXAutoscaleClusterRequest) {
+        if (distroXAutoscaleClusterRequest.getTimeAlertRequests().isEmpty()) {
+            return false;
+        }
+        return !(cluster.getTimeAlerts().isEmpty() && distroXAutoscaleClusterRequest.getTimeAlertRequests().isEmpty());
```

Comment:
Will this line ever return anything other than a true ?

Reply:
If there is no schedule based policy assigned than timeAlerts in the distroXAutoscaleClusterRequest would be empty and it would return a false and in this case we would be enabling stopStartScaling.

Reply:
Line 282 will always return true as there is only one case where this function should return false and that is covered above.

Reply:
Removed